### PR TITLE
Fixed Issue#476: Mongoid-history Tracking

### LINKED ISSUE #476: Garden Crops should track history
From @roryaronson:

> This sounds like some sort of 'Plant History'. Like, for every plant in a garden, it has the sow date and also the dates when it was in a different life stage?  That could be a great feature for long-term tracking of perennials, or for better gauging how long a stage will happen based on past plants in the garden. Am I understanding you correctly?

And from me:

> so right now it's a `tomato, sowed(date), current stage("Preparation")`  but next week it could be `tomato, sowed, "Juvenile"`. Right now we don't have a record of when that changed. But that could be fairly useful data-gathering wise.

### PR DESCRIPTION
**Pull request by TanSA05**
Using the `mongoid-history` gem [here](https://github.com/aq1018/mongoid-history/blob/v0.5.0/README.md) and @simonv3's help, I was able to track the number of changes in a `garden_crop` stage.

The display of `from` X stage `to` Y stage is yet to be implemented. However, for now this fixes #476

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -27,6 +27,11 @@ def update
   def show
     @user = User.find(params[:id])
     authorize @user
+    # Fetch history trackers
```
Comment (simonv3) [inline app/controllers/users_controller.rb]: Do we still need these within the show controller if we're using the API?

File: app/models/garden.rb [hunk 2]
```diff
@@ -24,6 +25,13 @@ class Garden
   embeds_many :pictures, cascade_callbacks: true, as: :photographic
   accepts_nested_attributes_for :pictures
 
+  track_history   on: :location,
+                  modifier_field: :modifier, # adds "belongs_to :modifier" to track who made the change, default is :modifier
```
Comment (simonv3) [inline app/models/garden.rb]: These are okay to ignore.

File: app/models/garden.rb [hunk 3]
```diff
@@ -24,6 +25,13 @@ class Garden
   embeds_many :pictures, cascade_callbacks: true, as: :photographic
   accepts_nested_attributes_for :pictures
 
+  track_history   on: :location,
```
Comment (simonv3) [inline app/models/garden.rb]: Will the location of a garden likely change? Does it make sense to track the history of the garden? Is it needed for this PR?

**Comment by simonv3** [comment]
Awesome! This looks great - clean too. I left some comments inline, wondering if some of the changes you did are still needed (do we need to track Gardens for this to work?). Let me know!

**Comment by TanSA05** [comment]
@simonv3 Made the changes you suggested in the comments. :)

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (TanSA05) [inline app/controllers/users_controller.rb]: No, I guess not. I removed it from the code.

File: app/models/garden.rb (hunk 3 above)
Comment (TanSA05) [inline app/models/garden.rb]: Nope, not tracking the garden for this issue. Just `garden_crop`

**Comment by simonv3** [comment]
:thumbup: Looks great.

**Pull request merged**